java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/Partition.java

```diff
@@ -1,40 +1,50 @@
 package com.godaddy.asherah.appencryption;
 
-public class Partition {
+public abstract class Partition {
 
   private final String partitionId;
   private final String serviceId;
   private final String productId;
 
-  public Partition(final String partitionId, final String serviceId, final String productId) {
+  protected Partition(final String partitionId, final String serviceId, final String productId) {
     this.partitionId = partitionId;
     this.serviceId = serviceId;
     this.productId = productId;
   }
 
+  /**
+   * Gets the partitionId for a partition
+   * @return The partition id
+   */
   String getPartitionId() {
     return partitionId;
   }
 
+  /**
+   * Gets the serviceId for a partition
+   * @return  The service id
+   */
   String getServiceId() {
     return serviceId;
   }
 
+  /**
+   * Gets the productId for a partition
+   * @return The product id
+   */
   String getProductId() {
     return productId;
   }
 
-  public String getSystemKeyId() {
-    return "_SK_" + serviceId + "_" + productId;
-  }
-
-  public String getIntermediateKeyId() {
-    return "_IK_" + partitionId + "_" + serviceId + "_" + productId;
-  }
-
-  @Override
-  public String toString() {
-    return getClass().getSimpleName() + "[partitionId=" + partitionId +
-      ", serviceId=" + serviceId + ", productId=" + productId + "]";
-  }
+  /**
+   * Gets the SystemKey Id associated with the partition
+   * @return The System Key Id
+   */
+  public abstract String getSystemKeyId();
+
+  /**
+   * Gets the Intermediate Key Id associated with the partition
+   * @return The IntermediateKey Id
+   */
+  public abstract String getIntermediateKeyId();
```

Comment:
get consistent on the javadoc comment. `SystemKey Id` or `System Key Id` as written in intermediate key?